packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -443,6 +446,9 @@ export const DatePickerMixin = (subclass) =>
     close() {
       if (this._overlayInitialized || this.autoOpenDisabled) {
         this.$.overlay.close();
+
+        // Avoid opening the virtual keyboard when the input gets re-focused on dropdown close
+        this.__setVirtualKeyboardEnabled(false);
```

Comment:
Should we use `_onVaadinOverlayClose` event listener instead? BTW, we might need to move it from `vaadin-date-picker` where it's defined to the mixin where it's added.

Reply:
Hmm....`_onVaadinOverlayClose` is invoked on "vaadin-overlay-close" event which doesn't guarantee the overlay is actually closing. The current `_onVaadinOverlayClose` actually [prevents the overlay from closing](https://github.com/vaadin/web-components/blob/master/packages/date-picker/src/vaadin-date-picker.js#L227) if the event is caused by an outside click on the field itself.

Reply:
Thanks, that makes sense.